Update to grpc-1.47 (and fix the connection-string format)

ptabor opened the pull request:
Minimal change required to upgrade to grpc-1.47. 

https://github.com/etcd-io/etcd/issues/14112. Connecting to `unix:localhost:port` instead of never officially supported `unix://localhost:port` (https://grpc.github.io/grpc/cpp/md_doc_naming.html)

ahrtr commented on api/go.mod:
```diff
@@ -9,7 +9,7 @@ require (
 	github.com/stretchr/testify v1.7.0
 	google.golang.org/genproto v0.0.0-20211118181313-81c1377c94b1
-	google.golang.org/grpc v1.41.0
+	google.golang.org/grpc v1.47.0
```
The title mentions 1.48, but actually 1.47

spzala replied:
:) I guess it's a typo in the PR title as the latest stable release is 1.47. lgmt.

ptabor replied:
Good catch. Reworded.

ahrtr commented:
@ptabor Overall the PR looks good to me. But I don't understand why do you add one more directory `out`, and get some go.mod and go.sum files included in it?  I guess they might be just intermediate files generated during your debug?

ptabor commented:
> @ptabor Overall the PR looks good to me. But I don't understand why do you add one more directory `out`, and get some go.mod and go.sum files included in it? I guess they might be just intermediate files generated during your debug?

Removed it. It was produces by intelij. I've just added to it to `.gitignore` to avoid such problems... and it has happened.

ahrtr reviewed (approved):
LGTM
